set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -56,40 +56,33 @@
 			produces = {HttpHeaders.CONTENT_TYPE_JSONLD_UTF8, HttpHeaders.CONTENT_TYPE_JSON_UTF8})
 	@ApiOperation(notes = SwaggerConstants.SAMPLES_JSONLD, value = "Create user set", nickname = "createUserSet", response = java.lang.Void.class)
 	public ResponseEntity<String> createUserSet(
-			@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
 			@RequestBody String userSet,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,			
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request)
 					throws HttpException {
 				
-		return storeUserSet(wskey, userSet, userToken, profile, request);
+		return storeUserSet(userSet, profile, request);
 	}
 	
 	/**
 	 * This method requests parsing of a user set in JsonLd format to a UserSet object
-	 * @param wsKey The API key
 	 * @param userSetJsonLdStr The user set in JsonLd format
-	 * @param userToken The user identifier
 	 * @param profile The profile definition
 	 * @param request HTTP request
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	protected ResponseEntity<String> storeUserSet(String wsKey, String userSetJsonLdStr, String userToken,
+	protected ResponseEntity<String> storeUserSet(String userSetJsonLdStr, 
 			String profileStr, HttpServletRequest request) throws HttpException {
 		try {
 			// validate user - check user credentials (all registered users can create) 
 			// if invalid respond with HTTP 401 or if unauthorized respond with HTTP 403;
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			Authentication authentication = verifyWriteAccess(Operations.CREATE, request);
```

Comment:
this call needs to be moved the calling method

Reply:
removed validateApiKey call